src/index.ts

```diff
@@ -114,86 +95,123 @@ async function updateToolbar(
   };
 
   const toolbar = getToolbar();
-
-  const isToolbarNotExist = toolbar.addItem(
-    'debugger-lifeCycle-button',
-    new ToolbarButton({
-      className: 'jp-DebuggerSwitchButton',
-      iconClassName: 'jp-ToggleSwitch',
-      onClick: async () => {
-        if (!debug.session.debuggedClients.delete(debug.session.client.path)) {
-          debug.session.debuggedClients.add(debug.session.client.path);
-          await debug.start();
-        } else {
-          await debug.stop();
-        }
-        toggleAttribute();
-      },
-      tooltip: 'Enable / Disable Debugger'
-    })
-  );
-
-  if (isToolbarNotExist && widget instanceof ConsolePanel) {
+  const itemAdded = toolbar.addItem('debugger-button', button);
+  if (itemAdded && widget instanceof ConsolePanel) {
     widget.insertWidget(0, toolbar);
   }
-  toggleAttribute();
 }
 
+/**
+ * A handler for debugging a widget.
+ */
 class DebuggerHandler<
   H extends ConsoleHandler | NotebookHandler | FileHandler
 > {
+  /**
+   * Instantiate a new DebuggerHandler.
+   * @param builder The debug handler builder.
+   */
   constructor(builder: new (option: any) => H) {
-    this.builder = builder;
+    this._builder = builder;
+  }
+
+  /**
+   * Dispose all the handlers.
+   * @param debug The debug service.
+   */
+  async disposeAll(debug: IDebugger) {
```

Comment:
Does this need to be `async`? Also, is this an appropriate place to use the `DisposableSet` from `@lumino/disposable`?

Reply:
Good question.

This method might not even be needed if we decide to drop the `close` command: https://github.com/jupyterlab/debugger/issues/295

Reply:
If you don't have any kernels that support debugging, you should probably never see a debugger icon/sidebar. It does seem like if you want your sidebar back as empty, you might be annoyed that you can't get rid of the debugger. I'm not sure which way to go about this. Since the right sidebar is empty by default, I'm ambivalent. What do you think?

Reply:
It's probably something we want to discuss in https://github.com/jupyterlab/debugger/issues/295.

But it does feel like having 2 "on / off" states puts some cognitive load on the user. Maybe the extension can be disabled via the extension manager to remove the sidebar?

Reply:
Perhaps there should be a show/hide toggle in the `View` menu that is *available* only when the debugger is in the sidebar and is *enabled* only if the debugger is not turned on.

You're right, it's multiple layers, but I think if it's in the view menu, it will both be discoverable and a little less confusing. It won't be an explicit thing in a user's field of vision.

Reply:
> is _enabled_ only if the debugger is not turned on.

Does "turned on" in that case mean "the toggle button is on"?

Reply:
To come back to the first comment, `disposeAll` doesn't need to be `async` indeed (removed in the latest commit).